### ISSUE TITLE
kool status - fatal error: concurrent map writes

### ISSUE THREAD

**Issue by fabriciojs** [kool-dev/kool#256]
**Describe the bug**

Sometimes` kool status` fails with what seems like a race condition on concurrent map writing.

```
$ kool status
fatal error: concurrent map writes
fatal error: concurrent map writes

goroutine 11 [running]:
runtime.throw(0xab8038, 0x15)
	/usr/local/go/src/runtime/panic.go:1116 +0x72 fp=0xc00006cc88 sp=0xc00006cc58 pc=0x4359f2
runtime.mapassign_faststr(0x9e80c0, 0xc000263b30, 0xab3d51, 0xe, 0x0)
	/usr/local/go/src/runtime/map_faststr.go:291 +0x3d8 fp=0xc00006ccf0 sp=0xc00006cc88 pc=0x4142d8
kool-dev/kool/cmd/shell.(*DefaultShell).LookPath(0xc00032e3c0, 0xb74780, 0xc0003035c0, 0x0, 0x0)
	/code/cmd/shell/shell.go:210 +0x13b fp=0xc00006cd50 sp=0xc00006ccf0 pc=0x85ccbb
kool-dev/kool/cmd/compose.(*DockerCompose).Args(0xc00032e410, 0xc0002237e0, 0x7f652d05c7d0, 0x0)
	/code/cmd/compose/docker-compose.go:59 +0x5e fp=0xc00006ce60 sp=0xc00006cd50 pc=0x85e73e
kool-dev/kool/cmd/shell.(*DefaultShell).Exec(0xc00032e230, 0xb747c0, 0xc00032e410, 0xc0002237e0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)
	/code/cmd/shell/shell.go:104 +0x49 fp=0xc00006cef8 sp=0xc00006ce60 pc=0x85bf49
kool-dev/kool/cmd.(*DefaultKoolService).Exec(...)
	/code/cmd/kool_service.go:106
kool-dev/kool/cmd.(*KoolStatus).getServiceInfo(0xc000332000, 0xc0001fba6d, 0x5, 0xc000280a80, 0xc0001f7c30)
	/code/cmd/status.go:170 +0x13f fp=0xc00006cfb8 sp=0xc00006cef8 pc=0x95cfff
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc00006cfc0 sp=0xc00006cfb8 pc=0x46a881
created by kool-dev/kool/cmd.(*KoolStatus).Execute.func1
	/code/cmd/status.go:86 +0x115

goroutine 1 [chan receive]:
kool-dev/kool/cmd.(*KoolStatus).Execute(0xc000332000, 0xf06aa8, 0x0, 0x0, 0xc00000e020, 0x7)
	/code/cmd/status.go:92 +0x23f
kool-dev/kool/cmd.DefaultCommandRunFunction.func1(0xc000322840, 0xf06aa8, 0x0, 0x0)
	/code/cmd/root.go:61 +0x18a
github.com/spf13/cobra.(*Command).execute(0xc000322840, 0xf06aa8, 0x0, 0x0, 0xc000322840, 0xf06aa8)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000a9080, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
kool-dev/kool/cmd.Execute(0xb74880, 0xf06aa8)
	/code/cmd/root.go:45 +0x45
main.main()
	/code/main.go:15 +0x70

goroutine 8 [semacquire]:
sync.runtime_Semacquire(0xc0001f7c38)
	/usr/local/go/src/runtime/sema.go:56 +0x45
sync.(*WaitGroup).Wait(0xc0001f7c30)
	/usr/local/go/src/sync/waitgroup.go:130 +0x65
kool-dev/kool/cmd.(*KoolStatus).Execute.func1(0xc000280a80, 0xc000260c00, 0x3, 0x4, 0xc000332000)
	/code/cmd/status.go:89 +0x13c
created by kool-dev/kool/cmd.(*KoolStatus).Execute
	/code/cmd/status.go:79 +0x1e5
```

**Kool version and environment**

```
$ docker-compose -v
docker-compose version 1.27.4, build 40524192
$ docker -v
Docker version 20.10.2, build 2291f61
$ cat /etc/*release | head -n1
CentOS Linux release 7.9.2009 (Core)
$ kool -v
kool version 1.9.0
```

**To Reproduce**

It is intermittent, it was the first time I faced it and could reproduce a couple o times, something like 20% of the attempts I made in this scenario,.

**Comment by fabriciojs** [comment]
The fix for this should be a matter of adding in either a channel or a `sync.Mutex` to coordinate the writing into the map linearly to avoid the error, making it thread-safe.

**Comment by danielsuguimoto** [comment]
FIxed. It will be available on 1.9.1
